cmd/tsbs_generate_queries/databases/timestream/devops_single_measure_test.go

```diff
@@ -0,0 +1,404 @@
+)
+func TestDevopsSingleMeasureGetMeasureNameWhere(t *testing.T) {
+	cases := []struct {
+		desc         string
+		measureNames []string
+		want         string
+	}{
+		{
+			desc:         "single measure",
+			measureNames: []string{"foo1"},
+			want:         "(measure_name = 'foo1')",
+		},
+		{
+			desc:         "multi host",
+			measureNames: []string{"foo1", "foo2"},
+			want:         "(measure_name = 'foo1' OR measure_name = 'foo2')",
+		},
+	}
+
+	for _, c := range cases {
+		b := BaseGenerator{SingleMeasure: true}
+		dq, err := b.NewDevops(time.Now(), time.Now(), 10)
+		if err != nil {
+			t.Fatalf("Error while creating devops generator")
+		}
+		d := dq.(*Devops)
+
+		if got := d.getMeasureNameWhereString(c.measureNames); got != c.want {
+			t.Errorf("%s: incorrect output: got %s want %s", c.desc, got, c.want)
+		}
+	}
+}
+
+func TestDevopsSingleMeasureGetSelectClausesAggMetrics(t *testing.T) {
+	cases := []struct {
+		desc    string
+		agg     string
+		metrics []string
+		want    string
+	}{
+		{
+			desc:    "single metric - max",
+			agg:     "max",
+			metrics: []string{"foo"},
+			want:    "max(case when measure_name = 'foo' THEN measure_value::double ELSE NULL END) as max_foo",
+		},
+		{
+			desc:    "multiple metric - max",
+			agg:     "max",
+			metrics: []string{"foo", "bar"},
+			want: "max(case when measure_name = 'foo' THEN measure_value::double ELSE NULL END) as max_foo," +
+				"max(case when measure_name = 'bar' THEN measure_value::double ELSE NULL END) as max_bar",
+		},
+		{
+			desc:    "multiple metric - avg",
+			agg:     "avg",
+			metrics: []string{"foo", "bar"},
+			want: "avg(case when measure_name = 'foo' THEN measure_value::double ELSE NULL END) as avg_foo," +
+				"avg(case when measure_name = 'bar' THEN measure_value::double ELSE NULL END) as avg_bar",
+		},
+	}
+
+	for _, c := range cases {
+		b := BaseGenerator{SingleMeasure: true}
+		dq, err := b.NewDevops(time.Now(), time.Now(), 10)
+		if err != nil {
+			t.Fatalf("Error while creating devops generator")
+		}
+		d := dq.(*Devops)
+
+		if got := strings.Join(d.getSelectClausesAggMetrics(c.agg, c.metrics), ","); got != c.want {
+			t.Errorf("%s: incorrect output: got %s want %s", c.desc, got, c.want)
+		}
+	}
+}
+
+func TestDevopsSingleMeasureGroupByTime(t *testing.T) {
+	expectedHumanLabel := "Timestream 1 cpu metric(s), random    1 hosts, random 1s by 1m"
+	expectedHumanDesc := "Timestream 1 cpu metric(s), random    1 hosts, random 1s by 1m: 1970-01-01T00:05:58Z"
+	expectedTable := "cpu"
+	expectedSQLQuery := `SELECT bin(time, 60s) AS minute,
+        max(case when measure_name = 'usage_user' THEN measure_value::double ELSE NULL END) as max_usage_user
+        FROM "db"."cpu"
+        WHERE (measure_name = 'usage_user') AND (hostname = 'host_9') AND time >= '1970-01-01 00:05:58.646325 +0000' AND time < '1970-01-01 00:05:59.646325 +0000'
+        GROUP BY 1 ORDER BY 1 ASC`
+
+	rand.Seed(123) // Setting seed for testing purposes.
```

Comment:
```suggestion
	rand.New(rand.NewSource(123))
```
Deprecated: As of Go 1.20